Fix error checking in docs deployment

JDawes-ScottLogic opened the pull request:
Perform the correct check for an error in the documentation deployment
script.

Fixes #278

BenLambertNcl reviewed (changes requested):
I have just realized I got it wrong in the gitter chat. I think it should be something like this:
```
  IF ($LASTEXITCODE -ne "0") {
    Write-Warning -Message 'Deploy Failed'
  }
```
As after reading into it a little bit, $error is an object containing an array of errors (so after the first error, will always return true), where as this uses the exit code of the last command.

It also needs changing inside `test-script.ps1`, as that will have the same problem

JDawes-ScottLogic commented:
I probably shouldn't have assumed that a half-remembered fix in gitter was 100% correct...

JDawes-ScottLogic committed:
Fix error checking in docs deployment

Perform the correct check for an error in the documentation deployment
script and test scripts.

BenLambertNcl reviewed (approved):
Thanks :smile:
